test/e2e/specs/editor/various/switch-to-draft.spec.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+/** @typedef {import('@playwright/test').Page} Page */
+/** @typedef {import('@wordpress/e2e-test-utils-playwright').Admin} Admin */
+/** @typedef {import('@wordpress/e2e-test-utils-playwright').RequestUtils} RequestUtils */
+
+test.use( {
+	switchToDraftUtils: async ( { page, admin, requestUtils }, use ) => {
+		await use( new SwitchToDraftUtils( { page, admin, requestUtils } ) );
+	},
+} );
+
+test.beforeAll( async ( { requestUtils } ) => {
+	await Promise.all( [
+		requestUtils.deleteAllPosts(),
+		requestUtils.deleteAllPages(),
+	] );
+} );
+
+test.afterAll( async ( { requestUtils } ) => {
+	await Promise.all( [
+		requestUtils.deleteAllPosts(),
+		requestUtils.deleteAllPages(),
+	] );
+} );
+
+test.describe(
+	'Clicking "Switch to draft" on a published/scheduled post/page',
+	() => {
+		[ 'schedule', 'publish' ].forEach( ( postStatus ) => {
+			[ 'large', 'small' ].forEach( ( viewport ) => {
+				[ 'post', 'page' ].forEach( ( postType ) => {
```

Comment:
This is a wild use for test loops, and I'm here for it 😄

Reply:
FWIW, it's what Playwright [recommended](https://playwright.dev/docs/test-parameterize) too 😆 .